backend/OKPluginVariableRendering/VariableRendering.cs

```diff
@@ -0,0 +1,326 @@
+﻿using Microsoft.Extensions.Logging;
+using Newtonsoft.Json.Linq;
+using Omnikeeper.Base.CLB;
+using Omnikeeper.Base.Entity;
+using Omnikeeper.Base.Entity.DataOrigin;
+using Omnikeeper.Base.Model;
+using Omnikeeper.Base.Utils.ModelContext;
+using Omnikeeper.Entity.AttributeValues;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace OKPluginVariableRendering
+{
+    public class VariableRendering : CLBBase
+    {
+        private readonly IRelationModel relationModel;
+        private readonly ICIModel ciModel;
+        private readonly IEffectiveTraitModel traitModel;
+        private readonly ILayerModel layerModel;
+        private readonly IAttributeModel attributeModel;
+        public VariableRendering(ICIModel ciModel, ILayerModel layerModel, IEffectiveTraitModel traitModel, 
+            IRelationModel relationModel, IAttributeModel attributeModel, ILatestLayerChangeModel latestLayerChangeModel) : base(latestLayerChangeModel)
+        {
+            this.relationModel = relationModel;
+            this.ciModel = ciModel;
+            this.traitModel = traitModel;
+            this.layerModel = layerModel;
+            this.attributeModel = attributeModel;
+        }
+
+        const int moduleGroupPrio = 10;
+        const int customersGroupPrio = 100;
+        const int networkSegmentGroupPrio = 1000;
+        const int networkInterfaceGroupPrio = 2000;
+        const int assignmentGroupDefaultPrio = 10000;
+        const int ciPrio = 100000;
+         
+
+        public override async Task<bool> Run(Layer targetLayer, JObject config, IChangesetProxy changesetProxy, IModelContext trans, ILogger logger)
+        {
+             logger.LogDebug("Start VariableRendering");
+
+            //return false;
+
+            Configuration cfg;
+
+            try
+            {
+                cfg = config.ToObject<Configuration>();
+                logger.LogDebug("Parsed configuration for VariableRendering.");
+            }
+            catch (Exception ex)
+            {
+                logger.LogError("An error ocurred while creating configuration instance.", ex);
+                return false;
+            }
+
+
+            foreach (var layer in cfg.InputLayerSet)
+            {
+                logger.LogDebug($"Started rendering variables for layer {layer}");
+
+                var layersetVariableRendering = await layerModel.BuildLayerSet(new[] { layer }, trans);
+
+                var allCIs = await ciModel.GetMergedCIs(new AllCIIDsSelection(), layersetVariableRendering, false, AllAttributeSelection.Instance, trans, changesetProxy.TimeThreshold);
+
+                var mainCIs = allCIs.Where(ci => ci.MergedAttributes.ContainsKey(cfg.BaseCI.RequiredTrait)).ToList();
```

Comment:
Hmm, not sure what you are doing here... the RequiredTrait means you need to check the CIs whether or not they have this trait. And only use CIs as "mainCIs" who fulfill that trait. Right now you're checking if the CI contains an attribute with the same name as `cfg.BaseCI.RequiredTrait`, which does not make sense, because that's a comparison between an attribute name and a trait ID.